assets/js/ajax_search_select.js

```diff
@@ -0,0 +1,168 @@
+$(function() {
+    "use strict";
+
+    $('.search-select-ajax').aplusSearchSelectAjax();
+});
+
+(function($) {
+    "use strict";
+
+    const pluginName = "aplusSearchSelectAjax";
+    var defaults = {
+        widget_selector: "#search-select-widget",
+        field_selector: 'input[type="text"]',
+        search_selector: '.dropdown-toggle',
+        result_selector: '.search-options',
+        selection_selector: '.search-selected',
+    };
+
+    function AplusSearchSelectAjax(element, options) {
+        this.element = $(element);
+        this.timeout = null;
+        if (this.element.prop("tagName") == "SELECT" && this.element.prop("multiple")) {
+            this.settings = $.extend({}, defaults, options);
+            this.init();
+        }
+    }
+
+    $.extend(AplusSearchSelectAjax.prototype, {
+
+        init: function() {
+            this.widget = $(this.settings.widget_selector).clone()
+                .removeAttr("id").removeClass("hide").insertBefore(this.element);
+            this.element.hide();
+            var self = this;
+            this.selection = this.widget.find(this.settings.selection_selector);
+            this.api_url = self.element.attr("data-search-api-url");
+            this.parameter_list = self.element.attr("data-key-parameter-list").split(",");
+            this.selection_li = this.selection.find("li").remove();
+            this.element.find("option:selected").each(function(index) {
+              $.ajax({
+                url: self.api_url + $(this).attr("value"),
+              }).done(function(data) {
+                self.addSelection(data['id'], self.parameter_list.map(
```

Comment:
It looks like the same `parameter_list.map(...).join(', ')` invocation is done in three places, so could you move it to a new small helper function in order to avoid repeating the same code?